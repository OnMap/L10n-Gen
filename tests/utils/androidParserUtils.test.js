const { reduceAndroidTranslationsPairs } = require('../../src/utils/androidParserUtils');

const parsedXml = [
  {
    value: 'הכל',
    key: { name: 'activity.filter.allPropertyPresentedOnTvProperties' }
  },
  {
    value: 'הצג',
    key: { name: 'activity.filter.property.count.show' }
  },
  {
    value: 'נכסים',
    key: { name: 'activity.filter.property.count.properties' }
  }
];

describe('Android translations file text parser utils', () => {
  it('reduce parsed xml to array of pairs', () => {
    expect(reduceAndroidTranslationsPairs(parsedXml)).toMatchSnapshot();
  });
});
