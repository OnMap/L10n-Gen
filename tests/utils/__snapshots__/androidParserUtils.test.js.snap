// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Android translations file text parser utils reduce parsed xml to array of pairs 1`] = `
Array [
  Array [
    "activity.filter.allPropertyPresentedOnTvProperties",
    "הכל",
  ],
  Array [
    "activity.filter.property.count.show",
    "הצג",
  ],
  Array [
    "activity.filter.property.count.properties",
    "נכסים",
  ],
]
`;
